src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.IO;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    internal static class StartupLogger
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/dotnet";
+
+        private static readonly string LogDirectory = GetLogDirectory();
+        private static readonly string StartupLogFilePath = LogDirectory != null ? Path.Combine(LogDirectory, "dotnet-tracer-loader.log") : null;
+
+        public static void Log(string message, params object[] args)
+        {
+            try
+            {
+                if (StartupLogFilePath != null)
+                {
+                    try
+                    {
+                        using (var fileSink = new FileSink(StartupLogFilePath))
+                        {
+                            fileSink.Info($"[{DateTime.UtcNow}] {message}{Environment.NewLine}", args);
+                        }
+
+                        return;
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+                }
+
+                Console.WriteLine(message, args);
+            }
+            catch
+            {
+                // ignore
+            }
+        }
+
+        public static void Log(Exception ex, string message, params object[] args)
+        {
+            message = $"{message}{Environment.NewLine}{ex}";
+            Log(message, args);
+        }
+
+        private static string GetLogDirectory()
+        {
+            string logDirectory = null;
+
+            try
+            {
+                var nativeLogFile = Environment.GetEnvironmentVariable("DD_TRACE_LOG_PATH");
+
+                if (!string.IsNullOrEmpty(nativeLogFile))
+                {
+                    logDirectory = Path.GetDirectoryName(nativeLogFile);
+                }
+
+                if (logDirectory == null)
+                {
+                    var windowsDefaultDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Datadog .NET Tracer", "logs");
+                    if (Directory.Exists(windowsDefaultDirectory))
+                    {
+                        logDirectory = windowsDefaultDirectory;
+                    }
+                    else
+                    {
+                        // either Linux or OS X
+                        if (Directory.Exists(NixDefaultDirectory))
+                        {
+                            logDirectory = NixDefaultDirectory;
+                        }
+                        else
+                        {
+                            try
+                            {
+                                Directory.CreateDirectory(NixDefaultDirectory);
```

Comment:
We only try to create the directory if we go down the Linux path. We should do this for Windows, too.